include/micm/solver/rosenbrock.hpp

```diff
@@ -189,7 +193,7 @@ namespace micm
     /// @param rate_constants List of rate constants for each needed species
     /// @param number_densities The number density of each species
     /// @param forcing Vector of forcings for the current conditions
-    void CalculateForcing(
+    virtual void CalculateForcing(
```

Comment:
is it necessary to make this virtual? it looks like you are using the extending class directly in the tests

Reply:
Well, when I didn't have virtual here, the functions I defined weren't called, I'll try removing this now and see if it breaks

Reply:
Yeah, removing `virtual` here broke the tests

Reply:
That's fine. I'll have to translate the HIRES and E5 equations though.

Reply:
oh, well maybe it's not worth the change then. we can just keep the virtual functions and always change later on if we want to